src/marqo/_httprequests.py

```diff
@@ -52,64 +64,106 @@ def send_request(
         with warnings.catch_warnings():
             if not self.config.cluster_is_remote:
                 warnings.simplefilter('ignore', InsecureRequestWarning)
-            try:
-                request_path = self.config.url + '/' + path
-                if isinstance(body, (bytes, str)):
-                    response = http_method(
-                        request_path,
-                        timeout=self.config.timeout,
-                        headers=req_headers,
-                        data=body,
-                        verify=to_verify
-                    )
-                else:
-                    response = http_method(
-                        request_path,
-                        timeout=self.config.timeout,
-                        headers=req_headers,
-                        data=json.dumps(body) if body else None,
-                        verify=to_verify
-                    )
-                return self.__validate(response)
-            except requests.exceptions.Timeout as err:
-                raise BackendTimeoutError(str(err)) from err
-            except requests.exceptions.ConnectionError as err:
-                raise BackendCommunicationError(str(err)) from err
+            
+            for attempt in range(max_retry_attempts + 1):
```

Comment:
It would be better to have more test cases for edge cases here in general, as the `send_request` method is used for almost everything. We want to make sure this code change is quite stable.